packages/dds/tree/src/simple-tree/api/configuration.ts

```diff
@@ -189,7 +189,10 @@ export class TreeViewConfiguration<
 			node: (schema) => markSchemaMostDerived(schema, true),
 			allowedTypes(types): void {
 				if (config.preventAmbiguity) {
-					checkUnion(types, ambiguityErrors);
+					checkUnion(
+						Array.from(types).map(({ type }) => type),
```

Comment:
This saves creating an extra array
```suggestion
						Array.from(types, ({ type }) => type),
```